db.go

```diff
@@ -437,22 +477,50 @@ func (db *DB) reload() (err error) {
 		db.metrics.reloads.Inc()
 	}()
 
+	goodBlocks, corruptedBlocks, err := db.openBlocks()
+	if err != nil {
+		return err
+	}
+	loadableBlocks, deleteableBlocks, err := db.loadableBlocks(goodBlocks, corruptedBlocks)
```

Comment:
goodBlocks , corruptedBlocks , deleteableBlocks, loadableBlocks : yeah the logic here is still a bit hard to follow.If you don't mind give me few days and I will think how to simplify.

Reply:
Haha, yeah, it's definitely a little hard to follow

Reply:
ping me when you address my other comments and I will think how to simplify that part again.